crypto/keypair.go

```diff
@@ -40,3 +45,22 @@ func DecodePrivateKey(in []byte, keytype KeyType) (priv PrivateKey, err error) {
 
 	return priv, err
 }
+
+var ss58Prefix = []byte("SS58PRE")
+
+// PublicKeyToAddress returns an ss58 address given a PublicKey
+// see: https://github.com/paritytech/substrate/wiki/External-Address-Format-(SS58)
+// also see: https://github.com/paritytech/substrate/blob/master/primitives/core/src/crypto.rs#L275
+func PublicKeyToAddress(pub PublicKey) common.Address {
+	enc := pub.Encode()
+	hasher, err := blake2b.New(64, nil)
+	if err != nil {
+		return ""
+	}
+	_, err = hasher.Write(append(ss58Prefix, enc...))
+	if err != nil {
+		return ""
+	}
+	checksum := hasher.Sum(nil)
+	return common.Address(base58.Encode(append(enc, checksum[:2]...)))
```

Comment:
`base58.Encode` returns a string and the `common.Address` type is `string`. Can we be more explicit about the `common.Address` type, ie use an imported `base58` type?

Reply:
could you clarify?

Reply:
:ok_hand: following our discussion